src/pages/NFTs/OpenBidNFT/BidModal.tsx

```diff
@@ -181,20 +197,76 @@ const MESSAGE = styled.div`
   }
 `
 
-export const BidModal: FC<{ setVisible: (x: boolean) => void; visible: boolean }> = (props) => {
-  const { setVisible, visible } = props
+export const BidModal: FC<{ setVisible: (x: boolean) => void; visible: boolean; details?: ISingleNFT }> = (props) => {
+  const { setVisible, visible, details } = props
   const [mode, setMode] = useState('bid')
+  const { sessionUser, fetchSessionUser } = useNFTProfile()
+  const { connected, publicKey } = useWallet()
+  const { bidOnSingleNFT } = useNFTCollections()
   const [isLoading, setIsLoading] = useState(false)
+  const [bidPrice, setBidPrice] = useState('000.000')
 
   const onCancel = () => setMode('bid')
   const reviewBid = () => setMode('review')
-  const confirmBid = () => {
-    setIsLoading(true)
-    // TODO: Fake API
-    setTimeout(() => {
+
+  useEffect(() => {
+    if (connected && publicKey) {
+      if (!sessionUser || sessionUser.pubkey !== `${publicKey}`) {
+        fetchUser()
+      }
       setIsLoading(false)
-      setVisible(false)
-      setMode('bid')
+    } else {
+      setIsLoading(false)
+      notify({
+        type: 'error',
+        message: (
+          <MESSAGE>
+            <div>Couldn't fetch user data, please connect your wallet and refresh this page.</div>
+          </MESSAGE>
+        )
+      })
+    }
+
+    return () => {}
+  }, [publicKey, connected])
+
+  const fetchUser = () => {
+    fetchSessionUser('address', `${publicKey}`).then((res) => {
+      if (!res || (res.response && res.response.status !== 200) || res.isAxiosError) {
+        notify({
+          type: 'error',
+          message: (
+            <MESSAGE>
+              <div>Couldn't fetch user data, please refresh this page.</div>
+            </MESSAGE>
+          )
+        })
+      }
+    })
+  }
+
+  const confirmBid = async () => {
+    setIsLoading(true)
+    const bidObject = {
+      clock: Date.now() + '', // string, not a number
+      tx_sig: 'RANDOM_TX_SIG_HERE',
+      wallet_key: 'WALLET_KEY_HERE',
+      auction_house_key: 'AUCTION_HOUSE_KEY_HERE',
+      token_account_key: 'RANDOM_TOKEN_ACCOUNT_KEY_HERE',
+      auction_house_treasury_mint_key: 'AUCTION_HOUSE_TREASURY_KEY_HERE',
+      token_account_mint_key: 'TOKEN_ACCOUNT_MINT_KEY_HERE',
+      buyer_price: 1.03 * Number(bidPrice) + '',
+      token_size: 'TOKEN_SIZE_HERE',
+      non_fungible_id: details.non_fungible_id,
+      collection_id: details.collection_id,
+      user_id: sessionUser?.user_id
+    }
+
+    const res = await bidOnSingleNFT(bidObject)
```

Comment:
For future reference, it is always good to wrap these kinds of calls in a `try { await call() } catch(err) {}` pattern.